core/server/data/migrations/versions/3.12/001-add-identity-permission.js

```diff
@@ -0,0 +1,54 @@
+const ObjectId = require('bson-objectid');
+const common = require('../../../../lib/common');
+
+module.exports.config = {
+    transaction: true
+};
+
+module.exports.up = async (options) => {
+    const connection = options.transacting;
+
+    const existingIdentityPermission = await connection('permissions').where({
+        action_type: 'read',
+        object_type: 'identity'
+    }).first();
+
+    if (existingIdentityPermission) {
+        common.logging.warn('Adding permission for read:identity');
+        return;
+    }
+
+    common.logging.info('Adding permission for read:identity');
+
+    await connection('permissions').insert({
+        id: ObjectId.generate(),
+        name: 'Read identities',
+        action_type: 'read',
+        object_type: 'identity',
+        created_at: new Date(),
+        created_by: 1,
+        updated_at: new Date(),
+        updated_by: 1
+    });
+};
+
+module.exports.down = async (options) => {
+    const connection = options.transacting;
+
+    const existingIdentityPermission = await connection('permissions').where({
+        action_type: 'read',
+        object_type: 'identity'
+    }).first();
+
+    if (!existingIdentityPermission) {
+        common.logging.warn('Removing permission for read:identity');
```

Comment:
```suggestion
        common.logging.warn('No read:identity permission found');
```

This would make a bit more sense I think?